app/controllers/registered_applications_controller.rb

```diff
@@ -0,0 +1,61 @@
+class RegisteredApplicationsController < ApplicationController
+  before_action :set_registered_application, only: [:show, :delete, :destroy]
+
+  def index
+    @registered_applications = current_user.registered_applications
+  end
+
+  def show
+    @events = @registered_application.events.group_by(&:name)
+  end
+
+  def new
+    @registered_application = RegisteredApplication.new
+  end
+
+  def create
+    @registered_application = RegisteredApplication.new(registered_application_params)
+    @registered_application.user = current_user
+
+    if @registered_application.save
+      flash[:notice] = "Your application was saved."
+      redirect_to action: "index"
+    else
+      @errors = @registered_application.errors
+      error_log = ""
+      @errors.full_messages.each do |error|
```

Comment:
This technique for showing errors does work, and it's cool that you figured out this way to do it. It's not the standard way that most apps display errors though, so I'd like you to review the Validations checkpoint in Bloccit, which walks you through the way most apps do it.